### DISCUSSION TITLE
インストール実績、登録陽性者数、濃厚接触者通知実績等の表示

### DISCUSSION THREAD

**Discussion by yusuke** [cocoa-mhlw/cocoa#254] [Ideas]
## その機能リクエストは何らかの問題に関連しますか / Is your feature request related to a problem?
COCOAのインストール数や陽性者による登録が伸び悩んでいること

## 解決策についてお書きください / Describe the solution you'd like
接触状況の表示画面で、インストール数や、登録陽性者数、濃厚接触者通知実績、さらにはインストール数がより伸びた場合の登録陽性者数、濃厚接触者通知数、そのうち実際に陽性が判明する数、命を救えるであろう人数などの想定を表示する。

国民の協力無しには実効性を発揮しづらいアプリです。

現在の利用状況に加え、たとえばあと「1割ユーザーが増えたらn十人の感染、n人の命を救える可能性がある」「5割ユーザーが増えたらn千人の感染、n百人の命を救える可能性がある」といった数字が表示されると協力してインストール数を増やそう、シェアしよう、インストールしてみようという動機に繋がるのではないかと思います。

シェアする際もアプリのリンクだけでなくそのテキストも含まれているといいですね。

## その他 / Additional context

![IMG_5539](https://user-images.githubusercontent.com/74894/123040347-94810680-d42e-11eb-89d3-573a4db3af9e.PNG)

![IMG_5541](https://user-images.githubusercontent.com/74894/123040386-a2cf2280-d42e-11eb-885b-6c6eef36c599.PNG)

**Comment by b-wind** [comment]
そのようなデータがあると広める根拠にもなるし、役にたつ実感がわきそうであると良いなと思います。


ただ現状ダウンロード数と陽性者登録数はCOCOAのページにあるとおりで、その他の数字は計測していないと言う認識ですね。
https://www.mhlw.go.jp/stf/seisakunitsuite/bunya/cocoa_00138.html


その種の値を計測出来る機能はプライバシーを守る名目ですべて削って有るはず。
（サーバーへのアクセス時にUAすら送っていない）

**Comment by yusuke** [comment]
アプリで計測すべきと言う話ではありません。

アプリの目的は「命を救うこと」でありアプリをリリースすることではありません。ですので目的の達成度合い、つまり実際どれくらいの命を救えたのか、救えなかった(インストールが増えれば救えた)のかをどこかで集計、推計しているはずです。

そのデータを費用対効果の確認やスマホ向け広告を流してインストールを増やすべきなのかと言った予算配分の参考にするだけでなく、アプリ内でも提示することで利用促進を図れるのではないかという提案です。

**Comment by b-wind** [comment]
> アプリの目的は「命を救うこと」でありアプリをリリースすることではありません。ですので目的の達成度合い、つまり実際どれくらいの命を救えたのか、救えなかった(インストールが増えれば救えた)のかをどこかで集計、推計しているはずです。

そう言うデータがあって、提供して貰えれば表示出来るかも知れませんね。（私も知りたいです）
過去にそのようなデータを開示できないかという議論はありました（ #66 ）

**Comment by Takym** [comment]
以前、最新情報ページ(#39, #149)を実装しましたが、このページで統計情報を告知するのは解決になりますでしょうか？（僕の技術力が足りない為、統計情報の取得・表示処理はまだ実装していません。）

**Comment by yusuke** [comment]
「アプリをインストールしていることが役に立っているんだ、もっとインストールが増えることで命を救えるんだ、シェアしよう、知人友人家族にもすすめよう」、という動機になる情報であれば改善になると思います。

**Comment by yusuke** [comment]
インストール数を増やすという目的を達成するのに最新の情報をフェッチして表示するというのは必ずしも必要ではありません。

ざっくり、インストール数が増えればこれだけの命が救えますよ、という言葉、数字でも効果は大きいと思います。

極端な話、昨年リリース直後から日本在住の人々全員がインストール、利用していたらどれくらい救えた、みたいな推計もどこかにあるのではないかと思います。

**Comment by Takym** [comment]
最新情報が必ずしも必要ではないのはその通りですが「この問題を既に解決できる実装がある」という事を報告した次第です。

**Comment by b-wind** [comment]
全くごもっともな意見だと思います。ただ、判断出来る立場では無いのでコラボレーターの方々からコメントを戴きたい所ですね。

**Comment by yusuke** [comment]
`最新情報が必ずしも必要ではないのはその通りですが「この問題を既に解決できる実装がある」という事を報告した次第です。`
ですね！
統計情報を元にどれくらい救えたのか皮算用・・もとい推定値を出すのも「インストールしていて良かった！」と思える材料として大きいと思います。

**Comment by kvaluation** [comment]
## A. ドイツの開示例
https://micb25.github.io/dka/index_en.html

2020年8月8日の私の読解例です。
https://twitter.com/info_kvaluation/status/1292027960308805632

5.1 shared legitimate diagnosis keys per day
陽性登録者のTEK の数の推移

(ドイツの他、アメリカ各州で色々開示が始まっていると想定できますが追えてません)

## B. うなぎたん。さんの可視化例
https://datastudio.google.com/u/0/reporting/7368173a-980a-42fd-9b6e-abc8bc495b26/page/WHOXC
データポータルにまとめられています。うなぎたんさん。日次から週次に変化した際も素早く対応してくださいました。

## C. 診断キーの統計データ等（接触日シート別冊）
https://datastudio.google.com/reporting/069598a2-3f01-4b51-b023-cdb478992182/page/wlBwB
データポータル４ページ目にて、毎日配信されているzip数や提供キーの数が開示されています。
（zipの重複ダウンロードや照合不足の不具合が一時期観測されたため、通知サーバーからの数を集計して公表してます）

## 意見
1. 統計データ、必ずしもアプリでなくとも、Webでも良いと考えます。アプリ未使用者に情報提供するならWebの方が効果的かと。
2. より重要なのは、陽性登録率を高めることです。技術的にアプローチできることがまだありそうに思われます。
3. また、ダウンロード数だけでなく、アクティブユーザー数を把握したいところです。

**Comment by keiji** [comment]
シェアをするときのテンプレート・テキストを工夫することでCOCOAのインストール率を高めて、利用者の母数を増やすことで陽性者の登録数（not 率）を向上させるという提案と理解しました。

ぼくからの提案ですが、もっとシンプルにシェアのテキストに「感染拡大防止のため、大切な人を守るために、接触確認アプリ『COCOA』をインストールしてください。アプリは次のURLからダウンロードできます」を追加するというのはどうでしょう。

----

#251 と合わせて、技術的な変更としては大きなものにはならないと考えています。一方で、テキストの内容をどうするかについては十分な検討が必要かと個人的には思います。

また、これもぼくの個人的な考えですが、例示されている中で「○人の命を救える可能性がある」というテキストはいわば死者数の予測があることが前提となります。COCOAは厚生労働省が配布しているもので、現時点で国としてそのような予測をしているのか。その予測はどのようなデータを元に、どのような手法を持って導き出されたものか（専門家の十分なレビューを得ているか）。可能性の提示に限定したとしても「救える」と言ってしまっていいのか。それぞれのポイントを整理する必要が出てきそうです。

とはいえ、「命を救える可能性…」についてはあくまで一例の1つと理解していて、そこについて議論を深めたいという意図はぼくにはありません。その上で、どんな数字であっても裏付けは必要ですし、これまで第４波まで経験してきた中で、刻一刻と変わる状況を把握して、アプリに最新の数字を反映（更新）していかなければなりません。
そういった機能を実装するのは難しくありません（簡単というわけでもありません）が、メンテナンスをしていかなければならないので、作業としては重いものになるとぼくは考えています。

**Comment by yusuke** [comment]
私の中でもまとまっていませんでした。

既存ユーザー向け:
1. アプリを使っていることで守られているという実感を得られるようにする(アプリのアンインストール、無効化を防ぐ)
2. 利用者が増えることでより効果が大きくなることを実感し、シェアボタンを押したくなる

潜在ユーザー向け:
3. シェア時の文面を工夫して、インストールをしてみようという気にさせる

といった施策が大事かなと思っての提案になります。

より詳しく:

1. 
COCOAはインストールして有効化しているだけでいいが、特に効果の実感は得にくいものです。
またこれまでの不具合や低い利用率の報道、ワクチン接種が進むことによる安心感の向上、バッテリーの減り、などからアンインストールをしようという動機が常にあります。
Twitterを見ても多くの人がアンインストールを検討していることが分かります。
https://twitter.com/search?q=cocoa+アンインストール&btnG=All&f=live

数字で示すのは1つの実施例ですが、インストールしている人が効果を上げていることが実感できれば引き続き継続して利用しようという動機になるのではないかと思います。

2.
利用者数は低くとどまっていますが、シェアしてしている人も残念ながら少ないです。
直近24時間でiPhoneからツイートしている人は5名、Androidは3名でした。
[iPhone利用者からのシェア](https://twitter.com/search?q=https%3A%2F%2Fapps.apple.com%2Fjp%2Fapp%2Fid1516764458%20source%3A%22Twitter%20for%20iPhone%22&src=typed_query&f=live)
[Android利用者からのシェア](https://twitter.com/search?q=https%3A%2F%2Fplay.google.com%2Fstore%2Fapps%2Fdetails%3Fid%3Djp.go.mhlw.covid19radar%20source%3A%22Twitter%20for%20Android%22&src=typed_query&f=live)

インストールが増えることで効果を上げられる、より多くの命が救えるのだということが実感できればシェアボタンを押す動機になるのではないかと思います。

3.
こちらは @keiji のまとめてくださった通りです。

焦点が曖昧なので1.、2.についてはissueを分けるのもいいかもしれませんね。

**Comment by yusuke** [comment]
商用のアプリであれば売上を最大化するために様々な施策を施したり、A/Bテストをしたりしてアクティブ率、インストール数の向上を目指します。

このアプリは命を救えるわけです。商用のアプリと同じく、またはそれ以上にそれなりのコスト、手間をかけてアクティブ率、インストール数の向上を目指してしかるべきなのかな、と思います。

COCOAの開発というスコープからははずれますが、もしこのアプリの効果測定、インストールベースが拡大した際にどれくらい効果が向上するのか(陽性者をどれくらい探し出せて、どれくらい感染拡大を防げて、どれくらい命を救えるのか）という試算がないのでしたら早急に試算すべきと思います。

**Comment by keiji** [comment]
@yusuke 

場合によってはDiscussionに移行した方が良いかもしれませんが、タイトルにあるとおり、アプリ内に「インストール実績、登録陽性者数、濃厚接触者通知実績等」を表示する点に絞って回答します。

すでに https://github.com/cocoa-mhlw/cocoa/issues/250#issuecomment-867070103 にも書いたとおり、推計や試算、どんな数字であっても裏付けは必要ですし、これまで第４波まで経験してきた中で、刻一刻と変わる状況を把握して、アプリに最新の数字を反映（更新）していかなければなりません。
「機能」としての実装は難しくありません（簡単というわけでもありません）が、メンテナンスをしていかなければならないので、作業としては重いものになるとぼくは考えています。

商用のアプリ開発について、インストール率やアクティブ率を上げていく施策の重要性については、ぼくもアプリ開発者ですので承知しています。一方、COCOAについては状況が異なり、基本的には入れっぱなしで、濃厚接触者の通知や陽性判定を受けるまでインストールしていることも忘れていることが理想だと考えています。

現時点でアプリとしては、利用者に安心して使い続けてもらえるように「アプリがきちんと動いていることが確認できること」と「動いていなかったときの対応について適切に案内すること」（ #252 ）が課題だと考えています（個人的には #106 のように動いていないときに通知する機能も欲しいですね）。

さらに言えば、アンインストールのきっかけになっている「バッテリーを異常に消費する現象 #248 」についても情報を集めて、可能であれば解決していきたいと考えています。

こういった課題が解決できたあとに、あらためて考えてみても良いかもしれませんね。
もちろん、議論や提案についてはウェルカムなので、本IssueをDiscussionに移すかどうかも含めてご意見を聞かせてください。

**Reply by Takym** [reply]
> すでに https://github.com/cocoa-mhlw/cocoa/issues/250#issuecomment-867070103 にも書いたとおり、推計や試算、どんな数字であっても裏付けは必要ですし、これまで第４波まで経験してきた中で、刻一刻と変わる状況を把握して、アプリに最新の数字を反映（更新）していかなければなりません。
「機能」としての実装は難しくありません（簡単というわけでもありません）が、メンテナンスをしていかなければならないので、作業としては重いものになるとぼくは考えています。

情報の提供は [GitHub](https://github.com/cocoa-mhlw) 上のリポジトリで行いオープンソースで管理するのはどうでしょうか？
この方法であれば、勿論正確な情報かどうか検証する必要はあると思いますが、様々な方からの情報更新の貢献が見込まれます。

**Comment by yusuke** [comment]
`商用のアプリ開発について、インストール率やアクティブ率を上げていく施策の重要性については、ぼくもアプリ開発者ですので承知しています。一方、COCOAについては状況が異なり、基本的には入れっぱなしで、濃厚接触者の通知や陽性判定を受けるまでインストールしていることも忘れていることが理想だと考えています。`
命を救うため、利用者を最大化できるよう、随時アクティブ率、インストール数の推移などを集計し、計画し、実行に移せているのであれば良いと思います。

そうではなく、「リリースしておけば自動的に利用者は伸びて、利用停止やアンインストールされることはないであろう」と放置されている状態であれば対策する必要があると思います。GitHub上で議論するべきことなのかどうかは判断致しかねますが。
何もしていない状況であればまずは厚生労働省においてインストールベース、アクティブ率拡大のための施策の骨子を練るのが良いのではないかと思います。

**Reply by Takym** [reply]
本題とは関係ありませんが、引用する際は `インラインコード` ではなく
> 引用

を使って頂けますと読み易く幸いです。

**Comment by yusuke** [comment]
議論を発散させてしまいました。申し訳ないです。

1. 「アプリを使っていることで守られているという実感を得られるようにする」
については #252 でカバーされていそうですね。

3. 「シェア時の文面を工夫して、インストールをしてみようという気にさせる」
は、#251 で巻き取りできそうですね。

本件は
2. 「利用者が増えることでより効果が大きくなることを実感し、シェアボタンを押したくなる」
に焦点を当ててDiscussionとするのはいかがでしょう。

**Comment by keiji** [comment]
ではDiscussionに移行します。
移行後、タイトルの調整などはできると思うのでよろしくお願いします！

**Comment by kvaluation** [comment]
全体を拝読し、下記のような整理はどうだろかと、感じました。

## 1. COCOAユーザーに月１回通知を送信する
- 大切な人を守るために、使い続けて欲しいことを伝える。
- 最新バージョンを通知し、ご自身のが最新かどうか確認してもらう。
- 厚生労働省Webサイトの統計情報ページに誘導し、陽性登録者数の推移などお知らせし、陽性に際して登録して頂く手順なども予め知っておいてもらう。
- 最新の照合日の調べ方をお知らせし、毎日動作しているかどうか確認する方法をお伝えする（できれば、OS側のタイムスタンプなどみなくてよいCOCOAで完結する方法）
- 統計情報は別途議論するが、陽性登録者数は週次ではなく日次、日次の提供キー総数も統計データとして提供したらどうでしょうか。このGitHubリポジトリ対象外の話題とは思いますが。

## 2. 未利用者への案内
- インストールをしてもらうためのランディングページ（Webページ）をつくる。厚労省内でもデジタル庁でもどこでも良い。
- ランディングページの作成は、ドイツ・イングランドなどを参考として、わかりやすい最小限のデータのみとする。
- データを示したいが、学術論文などの裏付けが欲しいが、大学研究室に予算をつけるなど積極的な働きかけをして頂くべき課題であり、COCOAの技術論では解決できない。

## 3. アクティブ率を上げる
- 上記月1回の通知などで、休眠化を防ぐことが考えられます。
- 通知サーバーへのアクセス数は、アクティブ率の良い指標になりえるので、統計情報として集約したうえでの公開が望ましいと思います。
